packages/launchpad/package.json

```diff
@@ -28,7 +28,7 @@
     "@packages/frontend-shared": "0.0.0-development",
     "@percy/cypress": "^3.1.0",
     "@purge-icons/generated": "0.8.1",
-    "@testing-library/cypress": "8.0.0",
+    "@testing-library/cypress": "BlueWinds/cypress-testing-library#119054b5963b0d2e064b13c5cc6fc9db32c8b7b5",
```

Comment:
Do we have to reference this version of the testing library for these changes to be successful?  If we don't upgrade, what is the results/errors? What can users expect if they don't upgrade?

Reply:
The current version of cypress-testing-library continues to work - it doesn't break or throw any errors.

However, our tests were failing with that version because of this branch's more diligent checking for DOM updates. Specifically, we run into the following situation with our app tests:

1. `cy.get(someElement).scrollTo(somePlace)`.
2. Cypress starts scrolling, and Vue begins to re-render a virtualized list as the page scrolls.
3. `cy.getElementByTestId(somethingOrOther).click()` begins to execute. The *page isn't stable yet*. Our virtualized list has a nice fade-in effect, looks super cool. But this means that *the DOM continues to re-render even after `getElementByTestId` has resolved.
4. .click() waits for the page to stabilize, and the link to be actionable. But it never will be; the subject is detached from the DOM, and we can't retry the **command** `getElementByTestId`.

Whoops, detached DOM error - exactly what this branch was trying to solve. ^^;;

If users don't upgrade, they'll run into the same sort of issues, where some tests using cypress-testing-library that previously passed now fail - probably mostly those that run commands on animating elements.

There are two solutions. The traditional workarounds for Detached DOM errors - adding otherwise unnecessary .should() or .wait() commands can be used, but the better solution is to move cypress-testing-library to use queries, in which case 'it all just works', more reliably and less detached-DOMy than ever before, which is what I've done in my fork.

I could add the workarounds to get our tests passing rather than point to the updated cypress-testing-library, it wouldn't be a huge lift. But I wasn't fond of the idea of adding workarounds for the very bug I aim to solve. ^^;;

Reply:
Do you have a PR open to cypress-testing-library for these changes? Do we need an issue to update these dependencies once this is released & that version of cypress-testing-library is released?

Reply:
Opened #24529 and #24530 to track that work.

Reply:
I think we should consider shipping the relevant parts of Testing Library as part of Cypress. It's pretty popular, will make upgrading less confusing, and avoid any more of these situations where we need an third party to update, and then for users to potentially update two dependencies (Cypress and Testing Library) in lockstep.

While a much needed improvement, the detached DOM changes are already going to be somewhat disruptive; anything we can do to minimize that is ideal.

I think this could also be useful for Cypress Studio, which uses the same selectors as [Cypress.SelectorPlayground](https://docs.cypress.io/api/cypress-api/selector-playground-api#Syntax). I think the `findByRole` and `findByLabel` etc from TL can be better than our `data-cy` and `data-testid` selectors (in some cases, anyway).

We should talk about this more internally, but I'm on board with this.